src/main/java/uk/gov/hmcts/cmc/claimstore/filters/DocumentsFilter.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.cmc.claimstore.filters;
+
+import uk.gov.hmcts.cmc.claimstore.idam.models.UserDetails;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.domain.models.ClaimDocument;
+import uk.gov.hmcts.cmc.domain.models.ClaimDocumentCollection;
+import uk.gov.hmcts.cmc.domain.models.ClaimDocumentType;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class DocumentsFilter {
+
+    private static List<ClaimDocumentType> defendantViewableDocsType = Arrays.stream(ClaimDocumentType.values())
+        .filter(type -> !type.equals(ClaimDocumentType.CLAIM_ISSUE_RECEIPT))
+        .collect(Collectors.toList());
+
+    private static List<ClaimDocumentType> claimantViewableDocsType = Arrays.stream(ClaimDocumentType.values())
+        .filter(type -> !type.equals(ClaimDocumentType.SEALED_CLAIM))
+        .collect(Collectors.toList());
+
+    private static Predicate<ClaimDocument> docsForDefendant = claimDocument -> defendantViewableDocsType.stream()
+        .anyMatch(docType -> claimDocument.getDocumentType().equals(docType));
+
+    private static Predicate<ClaimDocument> docsForClaimant = claimDocument -> claimantViewableDocsType.stream()
+        .anyMatch(docType -> claimDocument.getDocumentType().equals(docType));
+
+    private DocumentsFilter() {
+        // Do nothing constructor
+    }
+
+    public static Claim filterDocuments(Claim claim, UserDetails userDetails) {
+
+        if (userDetails.isCaseworker() || !claim.getClaimDocumentCollection().isPresent()) {
```

Comment:
We have a tech-debt ticket to substitute `!optional.isPresent()` with `optional.isEmpty()`; might as well get in the habit though
```suggestion
        if (userDetails.isCaseworker() || claim.getClaimDocumentCollection().isEmpty()) {
```

Reply:
Strange, in Optional class they are using `if (!isPresent()) {` most of the places. Looks like isEmpty was introduced later. Will incorporate it.